api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -217,17 +219,50 @@ public Encounter saveEncounter(Encounter encounter) throws APIException {
 				toAdd.add(newObs);
 			}
 		}
-		
+
+		// remove all obs in toRemove recursively from encounter
+		Set<Obs> seenIt = new LinkedHashSet<>();
```

Comment:
One more question, how can I check the Codacy and CodeCoverage things locally? So I can fix them before pushing it.